src/lib/core/ports/primary/create-conversation-primary-ports.ts

```diff
@@ -1,13 +1,11 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { type TCreateConversationErrorResponse, type TCreateConversationRequest, type TCreateConversationSuccessResponse } from "../../usecase-models/create-conversation-usecase-models";
+import { type TCreateConversationResponse, type TCreateConversationErrorResponse, type TCreateConversationRequest, type TCreateConversationSuccessResponse } from "../../usecase-models/create-conversation-usecase-models";
 
 export interface CreateConversationInputPort {
-    presenter: CreateConversationOutputPort<any>;
-    execute(request: TCreateConversationRequest): Promise<void>;
+    execute(request: TCreateConversationRequest): Promise<TCreateConversationResponse>;
 }
 
-export interface CreateConversationOutputPort<TResponse> {
-    response: TResponse;
+export interface CreateConversationOutputPort {
     presentSuccess(success: TCreateConversationSuccessResponse): void;
     presentError(error: TCreateConversationErrorResponse): void;
```

Comment:
Probably not a big deal, but shouldn't these be typed with `TCreateConversationSuccessViewModel` and `TCreateConversationErrorViewModel`, analogous to what you do with List Conversations and List Source Data? Or is it different because this is creating rather than listing? If so, would it be worth capturing that distinction in a comment?

Reply:
Yes indeed, I'm missing a return type as the implementation of these are supposed to return a VM. Although I'm a bit confused now. If we change this to use signals, then you're technically not returning anything (just generating a side effect, by updating the signal). I'll wait for @maany 's follow-up answer before deciding how to type this